src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -246,84 +360,91 @@ public void resetNeededResources()
     }
 
     /**
-     * Check if the builder requires a certain ItemStack for the current construction.
-     *
-     * @param stack the stack to test.
-     * @return true if so.
+     * Update the available resources (from the chest or builder's inventory)
+     * which are needed for the build
      */
-    public boolean requiresResourceForBuilding(ItemStack stack)
+    private void updateAvailableResources()
     {
-        return neededResources.containsKey(stack.getUnlocalizedName());
-    }
+        final EntityCitizen builder = getWorkerEntity();
+        resourcesAvailable.clear();
 
-    /**
-     * Provides a view of the builder building class.
-     */
-    public static class View extends AbstractBuildingWorker.View
-    {
-        private HashMap<String, Integer> neededResources;
 
-        /**
-         * Public constructor of the view, creates an instance of it.
-         *
-         * @param c the colony.
-         * @param l the position.
-         */
-        public View(final ColonyView c, final BlockPos l)
+        InventoryCitizen builderInventory = null;
+        if (builder!=null)
         {
-            super(c, l);
+            builderInventory = builder.getInventoryCitizen();
         }
 
-        /**
-         * Gets the blockOut Window.
-         *
-         * @return the window of the builder building.
-         */
-        @NotNull
-        @Override
-        public Window getWindow()
-        {
-            return new WindowHutBuilder(this);
-        }
 
-        @Override
-        public void deserialize(@NotNull ByteBuf buf)
+        for (@NotNull final Map.Entry<String, ItemStack> entry : neededResources.entrySet())
         {
-            super.deserialize(buf);
+            final ItemStack itemStack = entry.getValue();
+            BuildingBuilderResource resource = resourcesAvailable.get(entry.getKey());
+            if (resource == null)
+            {
+                resource = new BuildingBuilderResource(entry.getValue().getDisplayName(), itemStack,0,itemStack.getCount());
+            }
 
-            final int size = buf.readInt();
-            neededResources = new HashMap<>();
+            if (builderInventory!=null)
+            {
+                resource.setAvailable(resource.getAvailable() + InventoryUtils.getItemCountInInventory(builderInventory, entry.getValue().getItem(), entry.getValue().getItemDamage()));
+            }
 
-            for (int i = 0; i < size; i++)
+            final IInventory chestInventory = this.getTileEntity();
+            if (chestInventory!=null)
             {
-                final String block = ByteBufUtils.readUTF8String(buf);
-                final int amount = buf.readInt();
-                neededResources.put(block, amount);
+                resource.setAvailable(resource.getAvailable() + InventoryUtils.getItemCountInInventory(chestInventory, entry.getValue().getItem(), entry.getValue().getItemDamage()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Split this 182 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)